insights/parsers/tests/test_openstack_server_list_all.py

```diff
@@ -0,0 +1,61 @@
+ | ID                                   | Name         | Status | Networks               | Image          | Flavor     |
+ +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+ | 410b05bb-59b7-4b4c-88e9-975c811d68da | compute-0    | ACTIVE | ctlplane=192.168.24.20 | overcloud-full | compute    |
+ | f891e98b-4df6-4c90-9bf1-39cf8ac900b0 | compute-1    | ACTIVE | ctlplane=192.168.24.9  | overcloud-full | compute    |
+ | 3d62cd7e-41d2-43dd-a5bf-5935bc319fae | controller-0 | ACTIVE | ctlplane=192.168.24.10 | overcloud-full | controller |
+ +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+ """
+
+INPUT_FORAMT_WRONG = """
++--------------------------------------+--------------+--------+------------------------+----------------+------------+
+ | ID                                   | Name         | Status | Networks               | Image          | Flavor     |
+ +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+ | 410b05bb-59b7-4b4c-88e9-975c811d68da | compute-0    | ACTIVE | ctlplane=192.168.24.20 | overcloud-full | compute    |
+ | f891e98b-4df6-4c90-9bf1-39cf8ac900b0 | compute-1    | ACTIVE | ctlplane=192.168.24.9 | overcloud-full | compute      |x
+ +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+""".strip()
+
+
+def test_empty_openstack_server_list():
+    with pytest.raises(SkipException) as e_info:
+        OpenstackServerListAll(context_wrap(""))
+    assert "Empty content." in str(e_info.value)
+
+
+def test_openstack_server_list_options():
+    parser_result = OpenstackServerListAll(context_wrap(VALID_TABLE))
+    parser_data = parser_result.data
+    assert len(parser_data) == 3
+    assert parser_data[0] == {
+        'ID': '410b05bb-59b7-4b4c-88e9-975c811d68da',
+        'Name': 'compute-0',
+        'Status': 'ACTIVE',
+        'Networks': 'ctlplane=192.168.24.20',
+        'Image': 'overcloud-full',
+        'Flavor': 'compute'
+    }
+    assert parser_data[1]["Name"] == 'compute-1'
+    assert parser_data[1]["Image"] == 'overcloud-full'
+    assert parser_data[1]["Flavor"] == 'compute'
+    assert parser_data[0]["Networks"] == 'ctlplane=192.168.24.20'
+
+    res = OpenstackServerListAll(context_wrap(INPUT_FORAMT_WRONG))
+    assert len(res.bad_lines) == 1
+    assert res.bad_lines[0] == ' | f891e98b-4df6-4c90-9bf1-39cf8ac900b0 | compute-1    | ACTIVE | ctlplane=192.168.24.9 | overcloud-full | compute      |x'
+
+
+def test_openstack_server_list_all_documentation():
+    failed_count, tests = doctest.testmod(
+        openstack_server_list_all,
+        globs={'output': openstack_server_list_all.OpenstackServerListAll(context_wrap(VALID_TABLE))}
+    )
+    assert failed_count == 0
```

Comment:
This code actually tests the **Examples:** section of your docstring, which you don't currently have.